src/env.cpp

```diff
@@ -936,13 +936,16 @@ void env_stack_impl_t::push_nonshadowing() {
 
 void env_stack_impl_t::push_shadowing() {
     // Propagate local exported variables.
-    // TODO: this should take all local exported variables, not just those in the top scope.
     auto node = std::make_shared<env_node_t>(true, nullptr);
-    for (const auto &var : locals_->env) {
-        if (var.second.exports()) {
-            node->env.insert(var);
-            node->changed_exported();
+    auto cursor = locals_;
+    while (cursor) {
```

Comment:
Why is this not a `for` loop as well? Would save on the manual `cursor = cursor->next`.

Reply:
oh thanks, changed

Reply:
Did you forget to push? I'm still seeing the while-loop?

Reply:
yeah my bad, I pushed it now